src/backend/distributed/commands/rename.c

```diff
@@ -109,6 +109,12 @@ PreprocessRenameStmt(Node *node, const char *renameCommand,
 	 */
 	ErrorIfUnsupportedRenameStmt(renameStmt);
 
+	if (renameStmt->renameType == OBJECT_TABLE)
+	{
```

Comment:
would the same logic hold for indexes? (anything else?)

Reply:
I do not think we append shardId to types. So I can't think of a reason why a type rename should create deadlocks.

The 2 types that are created on table creation (such as events and _events[]) are covered by `OBJECT_TABLE`.

I should however test `OBJECT_FOREIGN_TABLE` especially with cstore tables

Reply:
After some testing, I added `OBJECT_FOREIGN_TABLE` support here. Also added some tests using `fake_fdw`